### PR TITLE
Validate Docs against Metadata Route

### PR DESCRIPTION
**Pull request by BarryShehadeh**
- Cleaned up and Enhanced "CheckMetadataOptions" Route to support multiple useful scenarios, and a world where not everything is perfect aka documented.

- There shouldn't be any breaking changes to the current flow of the route. There was a minor change to the flow in (ApiDoctor.Validation/Json/jsonresourcecollection.cs) where it seemed to be either a bug/typo or poor design to always overwrite options.

### CODE REVIEW

File: ApiDoctor.Console/ApiDoctor.ConsoleApp.csproj [hunk 1]
```diff
@@ -67,22 +67,26 @@
...
+    </Reference>
+    <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms, Version=2.23.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.23.302261847\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
+    </Reference>
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
+      <HintPath>..\packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
+    </Reference>
     <Reference Include="System" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Drawing" />
```
Comment (BarryShehadeh) [inline ApiDoctor.Console/ApiDoctor.ConsoleApp.csproj]: I'll clean up the extra refs before I merge.

File: ApiDoctor.Validation/OData/ODataParser.cs [hunk 2]
```diff
@@ -59,6 +60,7 @@ public class ODataParser
             { "Edm.SByte", -12 },
             { "Edm.DateTime", "2014-01-01T00:00:00Z" },
             { "Edm.DateTimeOffset", "2014-01-01T00:00:00Z" },
+            { "Edm.Duration", "P3DT1H" },
```
Comment (chackman) [inline ApiDoctor.Validation/OData/ODataParser.cs]: Please provide a more complete example (including minutes and seconds in this case)

File: ApiDoctor.Validation/OData/ODataParser.cs [hunk 3]
```diff
@@ -187,49 +189,66 @@ public static async Task<EntityFramework> ParseEntityFrameworkFromFileAsync(stri
...
         {
             Dictionary<string, object> propertyExamples = new Dictionary<string, object>();
 
             if (!string.IsNullOrWhiteSpace(ct.Namespace))
             {
-                propertyExamples.Add("@odata.type", $"{ct.Namespace}.{ct.TypeIdentifier}");
+                var metadataNamespace = metadataValidationConfigs?.ModelConfigs?.AliasNamespace ?? ct.Namespace;
+                var resourceTypeIdentifier = !string.IsNullOrEmpty(metadataValidationConfigs?.ModelConfigs?.AliasNamespace)
+                    ? $"{metadataNamespace}.{ct.TypeIdentifier}"
+                    : ct.TypeIdentifier;
```
Comment (chackman) [inline ApiDoctor.Validation/OData/ODataParser.cs]: Missing concatenation with `ct.Namespace`?  (shouldn't it be `ct.Namespace + "." + ct.TypeIdentifier` like in `ResourceDefinitionFromType`?)

Comment (chackman) [reply ApiDoctor.Validation/OData/ODataParser.cs]: If that's the case also make this code more consistent with `ResourceDefinitionFromType` - actually you could have a helper function that generates a type identifier from metadataValidationConfigs + default namespace + type name.

**Pull request merged**